modules/mbedtls/CMakeLists.txt

```diff
@@ -1,71 +1,210 @@
+#
+# Copyright (c) 2023 Linaro Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
 if(CONFIG_MBEDTLS)
-zephyr_interface_library_named(mbedTLS)
 
-if(CONFIG_MBEDTLS_BUILTIN)
-  target_compile_definitions(mbedTLS INTERFACE
-    MBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}"
-  )
+  if(CONFIG_MBEDTLS_BUILTIN)
+    if(CONFIG_MBEDTLS_ZEPHYR_ENTROPY AND NOT CONFIG_ENTROPY_HAS_DRIVER)
+    message(WARNING "No entropy device on the system, using fake entropy source!")
+    endif()
 
-  target_include_directories(mbedTLS INTERFACE
-    ${ZEPHYR_CURRENT_MODULE_DIR}/include
-    # mbedTLS v3.1.0 has psa_crypto_cipher.c including an internal header using <>.
-    # This line ensures the header can be found in Zephyr.
-    # When updating to mbedTLS > v3.1.0, please check if this has been corrected,
-    # and if so remove this include path.
-    ${ZEPHYR_CURRENT_MODULE_DIR}/library
-    configs
-    include
-  )
+    # Library for mbed TLS includes and configurations
+    add_library(mbedTLS INTERFACE)
 
-  zephyr_library()
+    # Add the config-file entry point
+    target_compile_definitions(mbedTLS
+      INTERFACE
+        MBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}"
+    )
 
-  file(GLOB
-    mbedtls_sources # This is an output parameter
-      ${ZEPHYR_CURRENT_MODULE_DIR}/library/*.c
+    # Add regular includes
+    target_include_directories(mbedTLS
+      INTERFACE
+        ${ZEPHYR_CURRENT_MODULE_DIR}/include
+        ${ZEPHYR_CURRENT_MODULE_DIR}/include/mbedtls
+        ${ZEPHYR_CURRENT_MODULE_DIR}/include/psa
+        ${ZEPHYR_CURRENT_MODULE_DIR}/library
+        configs
+        include
     )
 
-  if(CONFIG_MBEDTLS_DEBUG_EXTRACT_BASENAME_AT_BUILDTIME)
-    zephyr_cc_option(-fmacro-prefix-map=${ZEPHYR_CURRENT_MODULE_DIR}/library/=)
-  endif()
+    # Add base library with files required by all drivers/backends.
+    zephyr_library_named(mbedTLSBase)
 
-  zephyr_library_sources(
-    zephyr_init.c
-    ${mbedtls_sources}
-  )
+    # Base mbed TLS files
+    list(APPEND mbedtls_base_src
+      ${ZEPHYR_CURRENT_MODULE_DIR}/library/asn1parse.c
+      ${ZEPHYR_CURRENT_MODULE_DIR}/library/asn1write.c
```

Comment:
Why is this being added to a list instead of directly provided to zephyr_library_sources if its only use is being provided to zephyr_library_sources?

Reply:
Make the CMake more readable and give a clear source segregation base name as all are from the same library directory.
Please let me know, is any particular reason you want to add under directly to zephyr_library_sources